test/IRGen/dynamic_cast_functions.swift

```diff
@@ -0,0 +1,44 @@
+// RUN: %target-run-simple-swift | %FileCheck %s
+// REQUIRES: executable_test
+
+let a: ((Int, Int)) -> Void = { x in }
+let b: Any = a
+
+let cast1 = b as? (Int, Int) -> Void
+let cast2 = b as? ((Int, Int)) -> Void
+
+// CHECK: ok
+print((cast1 == nil) ? "ok" : "fail")
+// CHECK: ok
+print((cast2 != nil) ? "ok" : "fail")
+
+let c: () -> Void = { }
+let d: (()) -> Void = { x in }
+
+let e: Any = c
+let f: Any = d
+
+let cast3 = e as? () -> Void
+let cast4 = e as? (()) -> Void
+let cast5 = f as? () -> Void
+let cast6 = f as? (()) -> Void
+
+// CHECK: ok
+print((cast3 != nil) ? "ok" : "fail")
+// CHECK: ok
+print((cast4 == nil) ? "ok" : "fail")
+// CHECK: ok
+print((cast5 == nil) ? "ok" : "fail")
+// CHECK: ok
+print((cast6 != nil) ? "ok" : "fail")
+
+let g: (Int, Int) -> Void = { x, y in }
+let h: Any = g
+
+let cast7 = h as? ((Int, Int)) -> Void
+let cast8 = h as? (Int, Int) -> Void
+
+// CHECK: ok
+print((cast7 == nil) ? "ok" : "fail")
+// CHECK: ok
```

Comment:
I think `assert` (or StdlibUnitTest) checks are preferred to FileCheck'ing the output nowadays. Is that right, @airspeedswift  ?

Reply:
That's what I found tests in `test/Interpreter/` do for dynamic casts...

Reply:
Yeah, it's fine. Ignore my comment :)